mantidproject/mantid
# Updates to Engg algorithms: mask pulses, normalization, additional properties, file extensions, phase information

Introduces several modifications to the Engg algorithms to finish all the workflow plans discussed with scientists so far. All these are done at once to update the unit/doc/system tests for all the changes consistently:
- # 14788: Masking of some pulses, with defaults for EnginX
- # 13045: Add input property for rebinning bin size (for the initial rebinning in ToF) in EnggCalibrateFull
- # 13378: Normalization by proton charge always enabled by default in  EnggFocus (the Vanadium updates were done already)
- # 15338: use ".prm" as extension for the output calibration files, and  be able to generate individual files per bank

And it also adds several phase information files as a resource for various Engg scripts and algorithms.

This depends on:
- [x] #16092 needs to be merged first

**To test:**
- check documentation updates, now EnggFocus has two more child algorithms.
  Check the documentation of the new properties
- check code
- check the updates to the documentation of the Engg GUI
- unit/doc/system tests should all pass (one doc test was fitting a
  spurious peak from a pulse of Engin-X and has been updated)
- this should not change how the Engg GUI works (you will only notice some changes if
  the sample data had big pulses). In particular, calculate a normal calibration and
  see that the output ".prm" instrument files for GSAS are produced. Calculate a
  "cropped" calibration and see that the file is also produced correctly.
- make sure that if you specify a custom name in "cropped"
  calibration, next to the output "_all_banks.prm" there is a prm file
  with the name of the custom bank
- Some 20 EXPECT_CALL conditions were not being satisfied in
  EnggDiffractionPresenterTest (can be seen as warnings when running the test with
  `-V`). These have been fixed, and `VerifyAndClearExpectations` calls
  have been added to enforce those conditions as test failures rather than simply warnings.

Fixes #14788
Fixes #13378
Fixes #13045
Fixes #15338

Release notes: https://github.com/mantidproject/mantid/commit/492ad64aa8a5e5b7eec8eb6bd2ac6f09d90467e7

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
